### PR TITLE
i#4318 xarch memtrace: Support gzipped .raw files

### PR DESCRIPTION
**Pull request by derekbruening**
Adds a new gzip wrapper around std::istream to support compressing
.raw files.  This lets us easily shrink checked-in test files, and
aids other relocate-then-process situations with raw files.

Updates the altbin test by compressing its .raw files.

Issue: #4318

### CODE REVIEW

**Comment by derekbruening** [comment]
So once again Travis isn't triggering on a chained PR (compare vs base PR #4331 instead of master): I'm still not seeing what in the Travis config is causing this.  I'll re-base to master once #4331 (and its base #4330) is in and ensure Travis is green prior to merge.

**Comment by derekbruening** [comment]
Wow: so that made two show up (avoiding the dup is what the .travis.yml exclusion is there for).  Makes it seem like there's some Github PR trigger problem?  And any new commit to the new PR's branch will properly trigger Travis?

**Comment by derekbruening** [comment]
Nope: theory was wrong.  Reinstating the exclusion resulted in zero Travis runs.  Hmm.

File: clients/drcachesim/tracer/raw2trace_directory.cpp [hunk 1]
```diff
@@ -108,17 +109,35 @@ raw2trace_directory_t::open_thread_log_file(const char *basename)
...
     }
     NULL_TERMINATE_BUFFER(path);
-    in_files_.push_back(new std::ifstream(path, std::ifstream::binary));
+    std::istream *ifile;
+#ifdef HAS_ZLIB
+    if (is_gzipped)
+        ifile = new gzip_istream_t(path);
+    else
+#endif
+        ifile = new std::ifstream(path, std::ifstream::binary);
```
Comment (s-kanev) [inline clients/drcachesim/tracer/raw2trace_directory.cpp:139]: This statement hanging off the else in the #ifdef sounds like trouble. Maybe pull the declaration of is_gzipped out of the first ifdef and check it here again?

File: clients/drcachesim/common/gzip_istream.h [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* gzip_istream_t: a wrapper around zlib gzFile to match the parts of the
+ * std::istream interface we use for raw2trace and file_reader_t.
+ * Supports only limiting seeking within the current internal buffer.
```
[the remark this answers is not in the document]
Comment (derekbruening) [reply clients/drcachesim/common/gzip_istream.h:35]: Done.

File: clients/drcachesim/tracer/raw2trace_directory.cpp (hunk 1 above)
Comment (derekbruening) [reply clients/drcachesim/tracer/raw2trace_directory.cpp:139]: SGTM.  Done.

**Pull request merged**